lib/parsers/string.js

```diff
@@ -1,21 +1,42 @@
 'use strict';
 
 const Iconv = require('iconv-lite');
+const LRU = require('lru-cache');
 
-exports.decode = function(buffer, encoding, start, end, options) {
+const decoderCache = new LRU({
+  max: 500,
+});
+
+exports.decode = function (buffer, encoding, start, end, options) {
   if (Buffer.isEncoding(encoding)) {
     return buffer.toString(encoding, start, end);
   }
 
-  const decoder = Iconv.getDecoder(encoding, options || {});
+  // Optimize for common case: encoding="short_string", options=undefined.
```

Comment:
I figure encoding is probably constant for most users. For my use case, it's always cesu8 (for utf8mb3).